default/AI/ShipDesignAI.py

```diff
@@ -1929,6 +1944,9 @@ def _get_design_by_name(design_name, verbose=False):
                 break
         if design:
             Cache.design_id_by_name[design_name] = design.id
+        else:
+            # invalid design
+            Cache.design_id_by_name[design_name] = EnumsAI.AIShipDesignTypes.SHIPDESIGN_INVALID
```

Comment:
Please add some ```print```-statement for logging/debugging here:

```print "Shipdesign %s seems not to exist: Caching as invalid design." % design_name``` or something similar.